src/shared/firewall-util.h

```diff
@@ -6,13 +6,20 @@
 
 #include "in-addr-util.h"
 
+typedef struct fw_ctx fw_ctx;
+
+int fw_ctx_new(fw_ctx **);
```

Comment:
please give return params names (and ideally the name `ret` here, which is what we typically do for the return parameter if there's only one)